docs/root/version_history/current.rst

```diff
@@ -121,6 +121,7 @@ New Features
 * http: added support for :ref:`proxy_status_config <envoy_v3_api_field_extensions.filters.network.http_connection_manager.v3.HttpConnectionManager.proxy_status_config>` for configuring `Proxy-Status <https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-proxy-status-08>`_ HTTP response header fields.
 * http: make consistent custom header format fields ``%(DOWN|DIRECT_DOWN|UP)STREAM_(LOCAL|REMOTE)_*%`` to provide all combinations of local & remote addresses for upstream & downstream connections.
 * http2: re-enabled the HTTP/2 wrapper API. This should be a transparent change that does not affect functionality. Any behavior changes can be reverted by setting the ``envoy.reloadable_features.http2_new_codec_wrapper`` runtime feature to false.
+* http3: add :ref:`early_data_option <envoy_v3_api_field_config.route.v3.RouteAction.early_data_option>` extension to allow upstream HTTP/3 sending requests over early data. If no extension is configured or :ref:`early_data_allows_safe_requests <envoy_v3_api_field_extensions.early_data_option.v3.DefaultEarlyDataOption.early_data_allows_safe_requests>` is configured true, HTTP/3 pool will send safe requests as early data to the host if the pool already cached 0-RTT credentials of that host. And if those requests fail and the underlying connection pool supports TCP fallback, the request may be retried automatically. If :ref:`early_data_allows_safe_requests <envoy_v3_api_field_extensions.early_data_option.v3.DefaultEarlyDataOption.early_data_allows_safe_requests>` is configured false, no requests are allowed to be sent as early data. Note that if any customized extension configures non-safe requests to be allowed over early data, the Envoy will not automatically retry them. If desired, config their retry manually. This feature requires both ``envoy.reloadable_features.conn_pool_new_stream_with_early_data_and_http3`` and ``"envoy.reloadable_features.http3_sends_early_data`` to be turned on.
```

Comment:
nit: "And if those" -> "If those"
nit: "them, unless they are explicitly configured to be retried." (perhaps with a link to where such config should be specified)